python/dpu_utils/ptutils/componenttrainer.py

```diff
@@ -150,6 +167,9 @@ def minibatch_iterator(data_iterator: Iterator[TensorizedData], return_partial_m
                     mb_loss.backward()
 
                     optimizer.step()
+                    if scheduler is not None:
+                        scheduler.step(epoch)

```

Comment:
Would it be helpful to change the interface to not only provide the `epoch`, but also the step index (would require an enumerate in the loop on L162)? This would make sense for things like warmup schedules and similar things, which are usually not epoch-based.